cpp/src/reshape/byte_cast.cu

```diff
@@ -31,36 +32,53 @@
 #include <thrust/iterator/constant_iterator.h>
 #include <thrust/iterator/counting_iterator.h>
 
+#include <type_traits>
+
 namespace cudf {
 namespace detail {
 namespace {
-struct byte_list_conversion {
-  /**
-   * @brief Function object for converting primitive types and string columns to lists of bytes.
-   */
-  template <typename T>
-  std::enable_if_t<!std::is_integral_v<T> and !is_floating_point<T>(), std::unique_ptr<column>>
-  operator()(column_view const&,
-             flip_endianness,
-             rmm::cuda_stream_view,
-             rmm::mr::device_memory_resource*) const
+
+template <typename T, typename Enable = void>
+struct byte_list_conversion_fn {
+  template <typename... Args>
+  static std::unique_ptr<column> invoke(Args&&...)
   {
     CUDF_FAIL("Unsupported non-numeric and non-string column");
   }
+};
 
+struct byte_list_conversion_dispatcher {
   template <typename T>
-  std::enable_if_t<is_floating_point<T>() or std::is_integral_v<T>, std::unique_ptr<column>>
-  operator()(column_view const& input_column,
-             flip_endianness configuration,
-             rmm::cuda_stream_view stream,
-             rmm::mr::device_memory_resource* mr) const
+  std::unique_ptr<column> operator()(column_view const& input,
+                                     flip_endianness configuration,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr) const
   {
-    size_type num_bytes = input_column.size() * sizeof(T);
+    return byte_list_conversion_fn<T>::invoke(input, configuration, stream, mr);
+  }
+};
+
+template <typename T>
+static constexpr bool is_supported()
+{
+  return cudf::is_numeric<T>() || std::is_same_v<T, cudf::string_view>;
+}
+
+template <typename T>
+struct byte_list_conversion_fn<
```

Comment:
Should this one be up with the other and not straddling the dispatcher?

Reply:
Sorry we still have to dispatch, since there are many types in `cudf::is_numeric`.